File: src/main/java/com/currencies/mainpackage/repositories/TicketRepository.java
package com.currencies.mainpackage.repositories;

import com.currencies.mainpackage.entities.Ticket;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.sql.Date;
import java.util.List;

public interface TicketRepository extends JpaRepository<Ticket, Integer> {
    @Query("from Ticket where fromPlace = :from AND toPlace = :to AND DATE(startFlightDate) = :startFlightDate")
    List<Ticket> findTickets(String from, String to, Date startFlightDate);
}

File: src/main/java/com/currencies/mainpackage/repositories/TicketRepository.kt
package com.currencies.mainpackage.repositories

import com.currencies.mainpackage.entities.Ticket
import com.currencies.mainpackage.entities.jpa.JpaCity
import java.sql.Date
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface TicketRepository : BaseRepository<Ticket, Long> {
    @Query("from Ticket where fromPlace = :from AND toPlace = :to AND DATE(startFlightDate) = :startFlightDate")
    fun findTickets(from: JpaCity, to: JpaCity, startFlightDate: Date): List<Ticket>
}
